mpifileutils.spec

```
@@ -0,0 +1,204 @@
+%else
+%global module_load() module load mpi/%{1}-%{_arch}
+%endif
+
+%if %{with_mpich}
+%global mpi_list mpich
+%endif
+%if %{with_openmpi}
+%global mpi_list %{?mpi_list} openmpi
+%endif
+%if %{with_openmpi3}
+%if 0%{?fedora}
+# this would be nice to use but causes issues with linting
+# since that is done on Fedora
+#{error: openmpi3 doesn't exist on Fedora}
+%endif
+%global mpi_list %{?mpi_list} openmpi3
+%endif
+
+%if (0%{?suse_version} >= 1500)
+%global mpi_libdir %{_libdir}/mpi/gcc
+%global mpi_lib_ext lib64
+%global mpi_includedir %{_libdir}/mpi/gcc
+%global mpi_include_ext /include
+%else
+%global mpi_libdir %{_libdir}
+%global mpi_lib_ext lib
+%global mpi_includedir  %{_includedir}
+%global mpi_include_ext -%{_arch}
+%endif
+
+%if (0%{?suse_version} >= 1500)
+%global cmake cmake
+%else
+%global cmake cmake3
+%endif
+
+Name:		mpifileutils
+Version:	0.10.1
+Release:	1%{?dist}
+Summary:	File utilities designed for scalability and performance.
+
+Group:		System Environment/Libraries
+License:	Copyright and BSD License
+URL:		https://hpc.github.io/mpifileutils
+Source:		https://github.com/hpc/%{name}/archive/v%{version}.tar.gz
+Patch1:     0.10.1...4ec7841.patch
+BuildRoot:  %_topdir/BUILDROOT
+%if (0%{?suse_version} >= 1500)
+BuildRequires: cmake >= 3.1
+BuildRequires: lua-lmod
+BuildRequires: libbz2-devel
+BuildRequires: libopenssl-devel
+%else
+BuildRequires: cmake3 >= 3.1
+BuildRequires: Lmod
+BuildRequires: bzip2-devel
+BuildRequires: openssl-devel
+%endif
+BuildRequires: libarchive-devel
+BuildRequires: gcc-c++
+
+%description
+File utilities designed for scalability and performance.
+
+%if %{with_openmpi3}
+%package openmpi3
+Summary:	File utilities designed for scalability and performance.
+BuildRequires: openmpi3-devel
+BuildRequires: dtcmp-openmpi3-devel
+BuildRequires: libcircle-openmpi3-devel
+
+%description openmpi3
+File utilities designed for scalability and performance.
+
+%if (0%{?suse_version} >= 1500)
+%package -n libmfu0-openmpi3
+Summary:	File utilities designed for scalability and performance.
+Provides: openmpi3-daos-0
+
+%description -n libmfu0-openmpi3
+Shared libraries for %{name}-openmpi3.
+%endif
+
+%package openmpi3-devel
+Summary:	File utilities designed for scalability and performance.
+%if (0%{?suse_version} >= 1500)
+Requires: libmfu0-openmpi3%{_isa} = %version-%release
+%else
+Requires: %{name}-openmpi3%{_isa} = %version-%release
+%endif
+
+%description openmpi3-devel
+Development files for %{name}-openmpi3.
+%endif
+
+%if %{with_mpich}
+%package mpich
+Summary:	File utilities designed for scalability and performance.
+BuildRequires: mpich-devel
+BuildRequires: dtcmp-mpich-devel
+BuildRequires: libcircle-mpich-devel
+Provides: mpich-daos-0
+
+%description mpich
+File utilities designed for scalability and performance.
+
+%if (0%{?suse_version} >= 1500)
+%package -n libmfu0-mpich
+Summary:	File utilities designed for scalability and performance.
+Provides: mpich-daos-0
```

```suggestion
Provides: %{name}-mpich-daos-0
```
Do the `Provides` need to explicitly prepend `name`?

---

Yes, you are correct there.  Thanks for the catch.